test/powershell/Language/Parser/Parser.Tests.ps1

```diff
@@ -649,48 +649,72 @@ foo``u{2195}abc
+            @{ Script = "14l";Expected = "14"; ExpectedType = [long] }
+            @{ Script = "-51l"; Expected = "-51"; ExpectedType = [long] }
+            #Tests for decimal notation
+            @{ Script = "15d";Expected = "15"; ExpectedType = [decimal] }
+            @{ Script = "0x14d";Expected = "0x14"; ExpectedType = [decimal] }
+            @{ Script = $([decimal]::MinValue); Expected = [decimal]::MinValue; ExpectedType = [decimal] }
+            @{ Script = "0x14d";Expected = "0x14"; ExpectedType = [decimal] }
+            @{ Script = "-25";Expected = "0x14"; ExpectedType = [decimal] }
+            @{ Script = $([decimal]::MinValue); Expected = [decimal]::MinValue; ExpectedType = [decimal] }
+            @{ Script = "0x15d";Expected = "349"; ExpectedType = [int] } # No decimal hex literals; d is a hex character
 			#Tests for exponential notation.
-			@{ Script = "0e0"; Expected = "0" }
-			@{ Script = "0e1"; Expected = "0" }
-			@{ Script = "1e2"; Expected = "100" }
-			@{ Script = $([int32]::MaxValue); Expected = $([int32]::MaxValue) }
-			@{ Script = "0e2"; Expected = "0" }
-			@{ Script = "-2e2"; Expected = "-200" }
-			@{ Script = "-0e2"; Expected = "0" }
-			@{ Script = "3e0"; Expected = "3" }
+			@{ Script = "0e0"; Expected = "0"; ExpectedType = [double] }
+			@{ Script = "0e1"; Expected = "0"; ExpectedType = [double] }
+			@{ Script = "1e2"; Expected = "100"; ExpectedType = [double] }
+			@{ Script = "0e2"; Expected = "0"; ExpectedType = [double] }
+			@{ Script = "-2e2"; Expected = "-200"; ExpectedType = [double] }
+			@{ Script = "2e-2"; Expected = "0.02"; ExpectedType = [double] }
+			@{ Script = "-0e2"; Expected = "0"; ExpectedType = [double] }
+			@{ Script = "3e0"; Expected = "3"; ExpectedType = [double] }
 			#Tests for floating point notation.
-			@{ Script = ".01"; Expected = "0.01" }
-			@{ Script = "0.0"; Expected = "0" }
-			@{ Script = "-0.1"; Expected = "-0.1" }
-			@{ Script = "9.12"; Expected = "9.12" }
-			@{ Script = $([single]::MinValue); Expected = $([float]::MinValue).ToString() }
-			@{ Script = $([float]::MaxValue); Expected = $([float]::MaxValue).ToString() }
+			@{ Script = ".01"; Expected = "0.01"; ExpectedType = [double] }
+        @{ Script = "0.0"; Expected = "0"; ExpectedType = [double] }
+			@{ Script = "-0.1"; Expected = "-0.1"; ExpectedType = [double] }
+			@{ Script = "9.12"; Expected = "9.12"; ExpectedType = [double] }
+			@{ Script = $([single]::MinValue); Expected = $([float]::MinValue).ToString(); ExpectedType = [single] }
+			@{ Script = $([float]::MaxValue); Expected = $([float]::MaxValue).ToString(); ExpectedType = [float] }
 			#Tests for the K suffix for numbers.
-			@{ Script = "0kb"; Expected = "0" }
-			@{ Script = "1kb"; Expected = "1024" }
-			@{ Script = "-2KB"; Expected = "-2048" }
+			@{ Script = "0kb"; Expected = "0"; ExpectedType = [int] }
+			@{ Script = "1kb"; Expected = "1024"; ExpectedType = [int] }
+			@{ Script = "-2KB"; Expected = "-2048"; ExpectedType = [int] }
```

Comment:
We don't test all code paths with suffixes. Ex.: 9.12Kb, -2lkb. Could you please enhance tests?

I suggest remove lines starting with "#Tests for the K suffix for numbers.". Then add test code like we did for expected type tests.
```powershell
Context "Numerical Notations Tests (starting at line 2374 to line 2452)" {
    BeforeAll {
$suffixes = @(
     @{ Name = ""; Value = 1 }
    @{ Name = "kb"; Value = 1024 }
    @{ Name = "MB"; Value = 1024*1024 }
    @{ Name = "Gb"; Value = 1024*1024*1024 }
    @{ Name = "Pb"; Value = 1024*1024*1024*1024 }
)
}

It "<Script> should return <Expected>, with type <ExpectedType>" -TestCases $testData {  
     param ( $Script, $Expected, $ExpectedType )  

foreach ($suffix in $Suffixes) {
    $NumWithSuffix = $Script + $suffix.Name
    $ExpectedNum = $Expected * $suffix.Value
    ExecuteCommand $NumWithSuffix | Should -BeOfType $ExpectedType
    ExecuteCommand $NumWithSuffix | Should -Be $ExpectedNum
}
}

Reply:
It's a good idea to do that, but since high values and high multipliers can result in overflow errors I think we're unfortunately a bit too likely to get failing tests by just iterating through the multipliers with everything (Pb overflows Int32, Gb overflows Int16, etc., etc.)

I've added a bunch, let me know if you think this is sufficient. 😄